packages/core/src/commands/proxy.ts

```diff
@@ -30,36 +31,49 @@ export function inspectRunningComposeApp(projectName: string) {
     const getNetworkName = (labels: string) => labels.split(',').map(l => l.split('=')).find(l => l[0] === 'com.docker.compose.network')?.[1]
     return Object.fromEntries(composeNetworks.map(x => ([getNetworkName(x.Labels), { name: x.Name }])))
   }
+
+  function parseJSONContainer(s: string) {
+    const ctr = JSON.parse(s) as {Names: string; ID: string; Labels: string }
+    return { names: ctr.Names, id: ctr.ID, labels: Object.fromEntries(ctr.Labels.split(',').map(l => l.split('='))) as Record<string, string> }
+  }
+
   const getPreevyAgentContainer = async () => {
-    const agentContainerId = await dockerCmd(`ps --filter ${projectFilter} --filter label=com.docker.compose.service=${agentServiceName} -q`)
-    if (!agentContainerId) {
+    const agentContainer = await dockerCmd(`ps --filter ${projectFilter} --filter label=com.docker.compose.service=${agentServiceName} --format json`)
+    if (!agentContainer) {
       return null
     }
-    return agentContainerId
+    return parseJSONContainer(agentContainer)
   }
 
   const getEnvId = async () => {
-    const agentContainerId = await getPreevyAgentContainer()
-    if (agentContainerId) {
-      return await dockerCmd(`inspect ${agentContainerId}  --format '{{ index .Config.Labels "preevy.env_id"}}'`)
-    }
-    return null
+    const agentContainer = await getPreevyAgentContainer()
+    return agentContainer?.labels['preevy.env_id']
+  }
+
+  const listAllContainers = async () => ((await dockerCmd(`ps -a --filter ${projectFilter} --format json`)).split('\n') ?? [])
```

Comment:
at this point, isn't it better to use dockerode?

Reply:
Yeah, I currently added this functionality, but when implementing it initially I use Dockerode.
There was a slight problem that it didn't understand docker contexts, by default it used:
unix:///var/run/docker.sock
But in docker for desktop, it changed in the last versions to:
unix:///{USER_HOME}/.docker/run/docker.sock
Which broke compatibility with the initial Dockerode defaults.
Back then, after playing it for a while and mange to make it work (reading config.json, and context files partially), I figured it's not worth the extra code and complexity at this point. (since we're already are using Docker cli)